modifications/DataClass.cc

```diff
@@ -172,14 +233,246 @@ void DataClass::uploadPlanToServer(QString location, QJsonObject plan){
+    QNetworkRequest request = QNetworkRequest(location);
+    request.setRawHeader("Authorization",QByteArray("Bearer ").append(accessToken));
+    QNetworkReply* reply = manager.get(request);
+    connect(reply,&QNetworkReply::finished, this, &DataClass::readyReadAllFlightPlans);
+}
+
+void DataClass::readyReadAllFlightPlans()
+{
+    QNetworkReply* reply = qobject_cast<QNetworkReply*>(sender());
+    QString replyStr = reply->readAll();
+    int statusCode = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if(statusCode == 201 || statusCode == 200){
+        QJsonDocument jsonReply = QJsonDocument::fromJson(replyStr.toUtf8());
+        QJsonArray jsonArray = jsonReply.array();
+        if(jsonArray.size() > 0){
+            flightData.clear();
+            for(int i=0; i<jsonArray.size(); i++){
+                addToFlightData(jsonArray[i].toObject());
+            }
+             emit getAllFlightPlansSuccessful();
+        }
+        else{
+            emit getAllFlightPlansFailed();
+        }
+    }else{
+        emit getAllFlightPlansFailed();
+        showMessage("Unable to get flight plans from management server", statusCode);
+    }
+}
+//*******************************************************************************************************************************************************************************
+void DataClass::getAllActivities()
+{
+    QString location = serverUrl + AerobridgeGlobals::getAllActivitiesUrl;
+    QNetworkRequest request = QNetworkRequest(location);
+    request.setAttribute(QNetworkRequest::FollowRedirectsAttribute, true);
+    request.setRawHeader("Authorization",QByteArray("Bearer ").append(accessToken));
+    QNetworkReply* reply = manager.get(request);
+    connect(reply,&QNetworkReply::finished, this, &DataClass::readyReadAllActivities);
+}
+
+void DataClass::readyReadAllActivities()
+{
+    QNetworkReply* reply = qobject_cast<QNetworkReply*>(sender());
+    QString replyStr = reply->readAll();
+    int statusCode = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if(statusCode == 201 || statusCode == 200){
+        QJsonDocument jsonReply = QJsonDocument::fromJson(replyStr.toUtf8());
+        QJsonArray jsonArray = jsonReply.array();
+        if(jsonArray.size() > 0){
+            activityID = jsonArray[0].toObject()["id"].toString();
+            emit getAllActivitiesSuccessful();
+        }else{
+            emit getAllActivitiesFailed();
+        }
+    }else{
+        emit getAllActivitiesFailed();
+        showMessage("Unable to get activites from management server", statusCode);
+    }
+}
+//*******************************************************************************************************************************************************************************
+// GET ALL OPERATORS
+void DataClass::getAllOperators()
+{
+    QString location = serverUrl + AerobridgeGlobals::getAllOperatorsUrl;
```

Comment:
these methods all on the inside look very similar, is there a smarter way to have a generic get method that takes the URL as an argument and another for its type of query and based on that you tie it to the identifiers for the slots/signals? all get and readyRead methods look too similar on the inside, unless control flow is different, maybe there's a smarter way to do this? 

@rhythmize @hrishiballal

Reply:
Yeah, there's quite some code duplication for preparing and sending the network request. It'll be better to use function arguments to pass these values and merge the these methods, maybe even for the appropriate slot to connect.

Reply:
something like `aerobridgeServerRequest(requestType, requestUrl, slotToConnect)` 
requestType can be GET/ POST/ PUT etc defined by an enum
You can always find a better name for method/ parameters :sweat_smile:

Reply:
@1AwesomeDeveloper thoughts on this?